osquery/tables/yara/yara_utils.cpp

```diff
@@ -99,7 +127,52 @@ Status compileSingleFile(const std::string& file, YR_RULES** rules) {
 
     if (result != ERROR_SUCCESS) {
       yr_compiler_destroy(compiler);
-      return Status(1, "Insufficient memory to get YARA rules");
+      return Status::failure("Insufficient memory to get YARA rules");
+    }
+  }
+
+  if (compiler != nullptr) {
+    yr_compiler_destroy(compiler);
+    compiler = nullptr;
+  }
+
+  return Status::success();
+}
+
+/**
+ * Compile yara rules from string and load it into rule pointer.
+ */
+Status compileFromString(const std::string& rule_defs, YR_RULES** rules) {
```

Comment:
The function `compileFromString` is similar to `compileSingleFile` and there are common code creating the yara compiler and error handling. I kept it as separate function to have different path for compiling the signature files and signature strings. Please let me know your thoughts.